common/src/main/java/org/valkyrienskies/mod/mixin/feature/ai/node_evaluator/SwimNodeEvaluatorMixin.java

```diff
@@ -0,0 +1,188 @@
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.valkyrienskies.mod.common.VSGameUtilsKt;
+import org.valkyrienskies.mod.common.config.VSGameConfig;
+
+@Mixin(SwimNodeEvaluator.class)
+public abstract class SwimNodeEvaluatorMixin extends NodeEvaluator {
+
+    //region Single block obstacle path type
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/BlockGetter;getFluidState(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/material/FluidState;"),
+        method = "getBlockPathType(Lnet/minecraft/world/level/BlockGetter;III)Lnet/minecraft/world/level/pathfinder/BlockPathTypes;"
+    )
+    private FluidState getFluidStateRedirectPathType(final BlockGetter instance, final BlockPos blockPos) {
+        final FluidState[] fluidState = {instance.getFluidState(blockPos)};
+        Level level = null;
+        if (!VSGameConfig.SERVER.getAiOnShips()) {
+            if (instance instanceof PathNavigationRegion) {
+                level = ((PathNavigationRegionAccessor) instance).getLevel();
+            } else if (instance instanceof Level) {
+                level = (Level) instance;
+            }
+            if (level != null && fluidState[0].isEmpty()) {
+
+                final double origX = blockPos.getX();
+                final double origY = blockPos.getY();
+                final double origZ = blockPos.getZ();
+                final Level finalLevel = level;
+                VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                    origY, origZ, 1,
+                    (x, y, z) -> {
+                        final BlockPos groundPos = new BlockPos(x, y, z);
+                        fluidState[0] = finalLevel.getFluidState(groundPos);
+                    });
+            }
+        }
+        return fluidState[0];
+    }
+
+    @Redirect(
+        at = @At(value = "INVOKE",
+            target = "Lnet/minecraft/world/level/BlockGetter;getBlockState(Lnet/minecraft/core/BlockPos;)Lnet/minecraft/world/level/block/state/BlockState;"),
+        method = "getBlockPathType(Lnet/minecraft/world/level/BlockGetter;III)Lnet/minecraft/world/level/pathfinder/BlockPathTypes;"
+    )
+    private BlockState getBlockStateRedirectPathType(final BlockGetter instance, final BlockPos blockPos) {
+        final BlockState[] blockState = {instance.getBlockState(blockPos)};
+        if (instance instanceof PathNavigationRegion && blockState[0].isAir() && VSGameConfig.SERVER.getAiOnShips()) {
+            final Level level = ((PathNavigationRegionAccessor) instance).getLevel();
+
+            final double origX = blockPos.getX();
+            final double origY = blockPos.getY();
+            final double origZ = blockPos.getZ();
+            VSGameUtilsKt.transformToNearbyShipsAndWorld(level, origX,
+                origY, origZ, 1,
+                (x, y, z) -> {
```

Comment:
Why are you using `transformToNearbyShipsAndWorld` if it only uses the last x,y,z that it gets and ignores the rest?